### PR TITLE
escape pod tgui

### PR DESCRIPTION
**Pull request by stanalbatross**
## About The Pull Request

added tgui for the escape pod control panel

as requested by shad0vvs

image
image
image

## Why It's Good For The Game

tgui good

## Changelog

:cl:
ui: converted the escape pod control console from nanoui to TGUI
/:cl:

### CODE REVIEW

File: code/modules/shuttles/escape_pods.dm [hunk 1]
```diff
@@ -214,50 +214,69 @@ As such, a new tracker datum must be constructed to follow proper child inherita
...
+/obj/structure/machinery/embedded_controller/radio/simple_docking_controller/escape_pod/ui_data(mob/user)
+	var/list/data = list()
+	var/launch_status[] = evacuation_program.check_launch_status()
+	var/datum/shuttle/ferry/marine/evacuation_pod/P = shuttle_controller.shuttles[id_tag]
+
+	data["docking_status"] = evacuation_program.dock_state
+	data["door_state"] = P.D.density
+	data["door_lock"] = P.D.locked
+	data["can_lock"] = evacuation_program.dock_state == (STATE_READY || STATE_DELAYED) ? TRUE : FALSE //FALSE
+	data["can_force"] = evacuation_program.dock_state == (STATE_READY || STATE_DELAYED) ? TRUE : FALSE
```
Comment (spookydonut) [inline code/modules/shuttles/escape_pods.dm:248]: redundant given you're already sending raw dock_state

Comment (stanalbatross) [reply code/modules/shuttles/escape_pods.dm]: done

File: code/modules/shuttles/escape_pods.dm [hunk 2]
```diff
@@ -214,50 +214,67 @@ As such, a new tracker datum must be constructed to follow proper child inherita
...
 	//TODO make sure you can't C4 this.
 
-	ex_act(severity)
-		return FALSE
+/obj/structure/machinery/embedded_controller/radio/simple_docking_controller/escape_pod/ex_act(severity)
+	return FALSE
+
+// TGUI stufferinos \\
+
+/obj/structure/machinery/embedded_controller/radio/simple_docking_controller/escape_pod/ui_interact(mob/user, ui_key, datum/nanoui/ui, force_open)
```
Comment (mullenpaul) [inline code/modules/shuttles/escape_pods.dm:222]: Shouldn't this be handled via attack_hand?

Comment (stanalbatross) [reply code/modules/shuttles/escape_pods.dm]: fixed

File: code/modules/shuttles/escape_pods.dm [hunk 3]
```diff
@@ -214,50 +214,69 @@ As such, a new tracker datum must be constructed to follow proper child inherita
...
+		return UI_CLOSE
+
+/obj/structure/machinery/embedded_controller/radio/simple_docking_controller/escape_pod/ui_data(mob/user)
+	var/list/data = list()
+	var/launch_status[] = evacuation_program.check_launch_status()
+	var/datum/shuttle/ferry/marine/evacuation_pod/P = shuttle_controller.shuttles[id_tag]
+
+	data["docking_status"] = evacuation_program.dock_state
+	data["door_state"] = P.D.density
+	data["door_lock"] = P.D.locked
```
Comment (Watermelon914) [inline code/modules/shuttles/escape_pods.dm:248]: these variable names suck so much, they're literally stored as this on the fucking object.

**Pull request merged**